pymc3/distributions/distribution.py

```diff
@@ -394,6 +396,13 @@ def _distr_parameters_for_repr(self):
         return []
 
 
+@_logprob.register(NoDistribution)
```

Comment:
Not sure about this one. Why did you add it?

I would guess an exception might be better

Reply:
Ah okay, I added that because the `BART` class wasn't dispatching according to `NoDistribution`s `logp` so had to add that separately like this.